src/plugins/discover/public/context_awareness/profile_providers/observability/observability_root_profile/accessors/get_app_menu.ts

```diff
@@ -0,0 +1,161 @@
+) => {
+  const dataQualityLocator =
+    share?.url.locators.get<DataQualityLocatorParams>(DATA_QUALITY_LOCATOR_ID);
+
+  if (dataQualityLocator) {
+    registry.registerCustomAction({
+      id: 'dataset-quality-link',
+      type: AppMenuActionType.custom,
+      controlProps: {
+        label: i18n.translate('discover.observabilitySolution.appMenu.datasets', {
+          defaultMessage: 'Data sets',
+        }),
+        testId: 'discoverAppMenuDatasetQualityLink',
+        onClick: ({ onFinishAction }) => {
+          const refresh = timefilter.getRefreshInterval();
+          const { from, to } = timefilter.getTime();
+
+          dataQualityLocator.navigate({
+            filters: {
+              timeRange: {
+                from: from ?? 'now-24h',
+                to: to ?? 'now',
+                refresh,
+              },
+            },
+          });
+
+          onFinishAction();
+        },
+      },
+    });
+  }
+};
+
+const registerCustomThresholdRuleAction = (
+  registry: AppMenuRegistry,
+  { data, triggersActionsUi }: ProfileProviderServices,
+  { dataView }: AppMenuExtensionParams
+) => {
+  registry.registerCustomActionUnderSubmenu(AppMenuActionId.alerts, {
+    id: AppMenuActionId.createRule,
+    type: AppMenuActionType.custom,
+    order: 101,
+    controlProps: {
+      label: i18n.translate('discover.observabilitySolution.appMenu.customThresholdRule', {
+        defaultMessage: 'Create custom threshold rule',
+      }),
+      iconType: 'visGauge',
+      testId: 'discoverAppMenuCustomThresholdRule',
+      onClick: ({ onFinishAction }) => {
+        const index = dataView?.toMinimalSpec();
+        const { filters, query } = data.query.getState();
+
+        return triggersActionsUi.getAddRuleFlyout({
+          consumer: 'logs',
+          ruleTypeId: OBSERVABILITY_THRESHOLD_RULE_TYPE_ID,
+          canChangeTrigger: false,
+          initialValues: {
+            params: {
+              searchConfiguration: {
+                index,
+                query,
+                filter: filters,
+              },
+            },
+          },
+          onClose: onFinishAction,
+        });
+      },
+    },
+  });
+};
+
+const registerCreateSLOAction = (
+  registry: AppMenuRegistry,
+  { data, discoverShared }: ProfileProviderServices,
+  { dataView, isEsqlMode }: AppMenuExtensionParams
+) => {
+  const sloFeature = discoverShared.features.registry.getById('observability-create-slo');
+
+  if (sloFeature) {
+    registry.registerCustomActionUnderSubmenu(AppMenuActionId.alerts, {
+      id: 'create-slo',
+      type: AppMenuActionType.custom,
+      order: 102,
+      controlProps: {
+        label: i18n.translate('discover.observabilitySolution.appMenu.slo', {
+          defaultMessage: 'Create SLO',
+        }),
+        iconType: 'bell',
+        testId: 'discoverAppMenuCreateSlo',
+        onClick: ({ onFinishAction }) => {
+          const index = dataView?.getIndexPattern();
```

Comment:
Do SLOs also not support ad hoc data views? I noticed in testing it wasn't passed through when creating an SLO.

Reply:
I just tried with an ad hoc data view (both kql and ES|QL) and it creates correctly the SLO from the pattern, can you help in reproducing this issue please?